examples/retrieve/tools/retriever_to_tool_example.py

```diff
@@ -0,0 +1,121 @@
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+"""
+Example demonstrating how to convert a retriever to a tool.
+
+This example shows:
+1. How to convert a custom StaticRetriever to a Tool
+2. How to define parameters for the tool
+3. How to execute the tool
+"""
+
+import neo4j
+from typing import Optional, Any, cast
+from unittest.mock import MagicMock
+
+from neo4j_graphrag.retrievers.base import Retriever
+from neo4j_graphrag.types import RawSearchResult
+from neo4j_graphrag.tools.tool import (
+    StringParameter,
+    ObjectParameter,
+)
+from neo4j_graphrag.tools.utils import convert_retriever_to_tool
+
+
+# Create a Retriever that returns static results about Neo4j
+# This would illustrate the conversion process of any Retriever (Vector, Hybrid, etc.)
+class StaticRetriever(Retriever):
+    """A retriever that returns static results about Neo4j."""
+
+    # Disable Neo4j version verification
+    VERIFY_NEO4J_VERSION = False
+
+    def __init__(self, driver: neo4j.Driver):
+        # Call the parent class constructor with the driver
+        super().__init__(driver)
+
+    def get_search_results(
+        self, query_text: Optional[str] = None, **kwargs: Any
+    ) -> RawSearchResult:
+        """Return static information about Neo4j regardless of the query."""
+        # Create formatted Neo4j information
+        neo4j_info = (
+            "# Neo4j Graph Database\n\n"
+            "Neo4j is a graph database management system developed by Neo4j, Inc. "
+            "It is an ACID-compliant transactional database with native graph storage and processing.\n\n"
+            "## Key Features:\n\n"
+            "- **Cypher Query Language**: Neo4j's intuitive query language designed specifically for working with graph data\n"
+            "- **Property Graphs**: Both nodes and relationships can have properties (key-value pairs)\n"
+            "- **ACID Compliance**: Ensures data integrity with full transaction support\n"
+            "- **Native Graph Storage**: Optimized storage for graph data structures\n"
+            "- **High Availability**: Clustering for enterprise deployments\n"
+            "- **Scalability**: Handles billions of nodes and relationships"
+        )
+
+        # Create a Neo4j record with the information
+        records = [neo4j.Record({"result": neo4j_info})]
+
+        # Return a RawSearchResult with the records and metadata
+        return RawSearchResult(records=records, metadata={"query": query_text})
+
+
+def main() -> None:
+    # Convert a StaticRetriever to a tool with specific parameters
+    static_retriever = StaticRetriever(driver=cast(Any, MagicMock()))
+
+    # Define parameters for the static retriever tool
+    static_parameters = ObjectParameter(
+        description="Parameters for the Neo4j information retriever",
```

Comment:
Is this description really required? It doesn't seem to have a real added-value.